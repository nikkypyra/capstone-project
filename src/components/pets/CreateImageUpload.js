import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import AddButton from '../buttons/AddButton'
import pawSrc from '../../images/taskpaw.png'

CreateImageUpload.propTypes = {
  previewImage: PropTypes.object.isRequired,
  handleImageUpload: PropTypes.func.isRequired,
  loading: PropTypes.number,
}

export default function CreateImageUpload({
  previewImage,
  handleImageUpload,
  loading,
}) {
  return (
    <>
      {previewImage.imageUrl ? (
        <ImageWrapper>
          <img src={previewImage.imageUrl} alt="" />
        </ImageWrapper>
      ) : (
        <ImageWrapper>
          <img src={pawSrc} alt="" />
        </ImageWrapper>
      )}
      <UploadWrapper>
        {loading > 0 && <span>Upload is {Math.trunc(loading)}% done</span>}
        <AddButton className="upload-button" text="Upload photo" />
        <input
          type="file"
          accept="image/*"
          className="file-input"
          onChange={handleImageUpload}
        />
      </UploadWrapper>
    </>
  )
}

const UploadWrapper = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  .file-input {
    opacity: 0;
    position: absolute;
  }
  button {
    margin-top: 8px;
  }
`

const ImageWrapper = styled.div`
  text-align: center;
  img {
    height: 200px;
    width: 200px;
    border: 8px solid var(--tertiary);
    border-radius: 50%;
  }
`
