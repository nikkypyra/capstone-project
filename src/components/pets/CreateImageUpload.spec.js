import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import CreateImageUpload from './CreateImageUpload'

afterEach(cleanup)

const previewImageMock = {}
const handleImageUpload = jest.fn()

describe('<CreateImageUpload />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <CreateImageUpload
          previewImage={previewImageMock}
          handleImageUpload={handleImageUpload}
        />
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
