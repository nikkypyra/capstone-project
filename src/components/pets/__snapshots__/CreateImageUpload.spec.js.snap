// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateImageUpload /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="CreateImageUpload__ImageWrapper-dfo38w-1 cLeDVg"
  >
    <img
      alt=""
      src="taskpaw.png"
    />
  </div>
  <section
    class="CreateImageUpload__UploadWrapper-dfo38w-0 bthBNR"
  >
    <button
      class="sc-AxjAm eirFEf"
    >
      Upload photo
    </button>
    <input
      accept="image/*"
      class="file-input"
      type="file"
    />
  </section>
</DocumentFragment>
`;
